src/state/user/updater.tsx

```diff
@@ -30,5 +31,13 @@ export default function Updater(): null {
     }
   }, [dispatch])
 
+  // keep locale in sync with the system if a locale is detected.
+  useEffect(() => {
```

Comment:
not sure why i did dark mode this way. we should by default always use the detected one (in dark mode's case, from url or system settings), and only set it in the store if the user sets it

Reply:
see the above, this code should not be in this updater thing. we should really delete the updater and move the above code that detects dark mode into the theme provider